### PR TITLE
Add save changes

### LINKED ISSUE #93: Use a Save Changes Button for Accounts page instead of saving on Update
Currently, if you change a field on the profile page it will automatically save the information. It would be nicer to have a button that sends all of the data at once instead of a couple little changes.

### LINKED ISSUE #94: Change Label and Value to Platform and Account
This is similar to how Rails auto defaults to using "Destroy" for removing a tuple. This is terminology that is familiar to the programmer, but we should make it sound more familiar to the user.

### LINKED ISSUE #98: Change the notification for successful profile edits.
Need to change alerts so that they do not shift the entire table down. Should have a static section of the page that will notify the user if their changes are saved to the database.

### PR DESCRIPTION
**Pull request by paytonshaltis**
Closes #93, closes #94, closes #98.

This PR refactors the existing profile page code in order to allow the user to click 'save changes' to apply updates to all modified accounts at once. This replaces the old functionality of just updating the modified field instantly.

There is client and server-side input validation. On the client side, buttons for adding a new account, canceling changes, and submitting changes all become disabled if the information is missing, reflecting similar behavior to most sites:

- The 'add account' button is disabled if there is an empty field in the add account portion of the form.
- The 'cancel changes' button is disabled if there have been no changes to the user's accounts.
- The 'submit changes' button is disabled if any of the accounts' textboxes are empty.

Functionality now works like so:

1. Anytime the user adds, updates, or deletes any user account, that change is stored on the client side in an appropriate array (add, edit, or remove). If the user makes a change to some text but then retypes it as it was originally, it is seen as no change, and thus the entry is _removed_ from the edit array. This makes sure that only changes to the user's account are actually sent back to the user.
2. If the user clicks submit, the server knows to either add, update, or delete each modified account. Accounts that were untouched by the user are thus untouched by the backend and the database.
3. If the user clicks cancel, they are prompted for confirmation and the page reloads. Any additions, updates, or deletes since the user clicked submit last are removed. This makes it possible to undo things like deletes.

Note that adding and deleting accounts are not finalized until you click 'submit.' Submitting will submit account additions, edits, and deletions, so the user will need to manually hit 'submit' no matter what change they want to make.

In addition, success and error messages are added to the page when the user is redirected back to it after submitting their changes, which closes that related issue mentioned above.

I definitely encourage you to test this functionality on your end a bit to make sure that there are not bugs, that buttons enable/disable at the right times, etc.

### CODE REVIEW

**Comment by sderacy** [comment]
Firstly, make sure you pull before you make any more changes @paytonshaltis 

Secondly, Yup. This is perfect. It's much more responsive to what the user does and pretty much addresses everything I think Nisha and I were talking about the other day.

I'm ready to approve this but first, I'd like to hear how @lelekaz @thadakn1 @matyis1 feel about this.

Obviously, now that it's much more reliant on the user to click through these, it's a little slower. I wonder if that should be considered a big deal, or if it's fine given the nature of how the user is expected to manage their accounts.

I can ask other people in the STEM building when I'm there later.

I'm gonna approve this but you should get their feedback too Payton. Really nice job!

You're an absolute unit.

File: routes/profile.js [hunk 1]
```diff
@@ -16,8 +16,14 @@ module.exports = function (app, path) {
    */
   app.get("/account/profile", isLoggedIn, (req, res) => {
     // Pass the user object to the profile page
+    const error = req.session.error;
```
Comment (sderacy) [inline routes/profile.js:19]: Good call.

**Review by sderacy — APPROVED** [review]
Good work again!

**Comment by thadakn1** [comment]
@paytonshaltis great work payton!!! it works perfectly!! 

just a minor suggestion, do you think on the frontend side we could move the "Add account" button outside of the div its in and add a <br> statement before so there's more spacing.  i tested out these changes locally and it would make it look better and still remain functional. it still works the way it should even if its outside of the div though which is good. just lmk what you think

**Comment by thadakn1** [comment]
> @paytonshaltis great work payton!!! it works perfectly!!
> 
> just a minor suggestion, do you think on the frontend side we could move the "Add account" button outside of the div its in and add a statement before so there's more spacing. i tested out these changes locally and it would make it look better and still remain functional. it still works the way it should even if its outside of the div though which is good. just lmk what you think

i can make a new issue for that if you just want to merge this and go right ahead. i can take up that issue too

**Comment by paytonshaltis** [comment]
> just a minor suggestion, do you think on the frontend side we could move the "Add account" button outside of the div its in and add a statement before so there's more spacing. i tested out these changes locally and it would make it look better and still remain functional. it still works the way it should even if its outside of the div though which is good. just lmk what you think

Yeah, that sounds like a good idea! I knew that the arrangement of buttons would likely need to change; I just wanted to get the functionality working correctly. If you want to pull from this branch, make the change, then push back here we can include these changes all in this same PR since I think they are relevant here.

**Comment by thadakn1** [comment]
perfect! i already pulled the branch so i'll make these changes permanent and i'll push back to this pr

**Comment by paytonshaltis** [comment]
@thadakn1 Changes to button placement look good!

**Pull request merged**